src/charm.py

```diff
@@ -679,6 +679,13 @@ def _install_repo_policy_compliance(self) -> bool:
         )
         return old_version != new_version
 
+    def enable_kernel_modules(self) -> None:
+        """Enable kernel modules needed by the charm."""
+        execute_command(["/usr/sbin/modprobe", "br_netfilter"])
+
+        with open("/etc/modules", "a", encoding="utf-8") as modules_file:
+            modules_file.write("br_netfilter\n")
```

Comment:
what about using: https://docs.python.org/3/library/pathlib.html#pathlib.Path.write_text?

Reply:
That function overwrite the file, and there does not seems to be a pathlib function for appending.
This current approach is trying to append to the end of the file rather than overwriting the file.